D/D.sublime-syntax

```diff
@@ -8,339 +8,2057 @@ file_extensions:
+    - include: not-whitespace-illegal
+  is-after-type:
+    - match: '\b({{name}})\b'
+      scope: variable.parameter.d
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - match: '\=\=|:'
+      scope: keyword.operator.logical.d
+      set: [is-after, type-specialization]
+  is-after:
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: function-argument-definition-list-content
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - include: not-whitespace-illegal
+  type-specialization:
+    - match: '\b(?:__vector|function|delegate|__parameters)\b'
+      scope: keyword.other.d
+      set:
+        - match: '(?=\))'
+          pop: true
+        - include: type-specialization
+    - match: '\breturn\b'
+      scope: keyword.control.flow.d
+      set:
+        - match: '(?=\))'
+          pop: true
+        - include: type-specialization
+    - match: '\bsuper\b'
       scope: variable.language.d
-    - match: |-
-        (?x)
-        (
-          (\b0(x|X)[0-9a-fA-F](_[0-9a-fA-F]+)*)            # 0x0, 0X0, 0x000_000
-          |
-          (
-            (\b[0-9]+(_[0-9]+)*\.?([0-9]+_[0-9]+)?[0-9]*)  # 0.0, 0_0.0, 0.0_0
-            |
-            (\.[0-9]+(_[0-9]+)*)                           # .0, .0_0
-          )((e|E)(\+|-)?[0-9]+)?                           # 0e+5
-        )
-        ([LlFfUuDd]|UL|ul)?                                # 0l, 0U, 0Lu, 0UL
-        \b
-      scope: constant.numeric.d
-    - include: string_escaped_char
-    - include: strings
-    - match: (==|!=|<=|>=|<>|<|>)
+      set:
+        - match: '(?=\))'
+          pop: true
+        - include: type-specialization
+    - match: '\b(?:struct|union|class|interface|enum)\b'
+      scope: storage.type.d keyword.declaration.d
+      set:
+        - match: '(?=\))'
+          pop: true
+        - include: type-specialization
+    - match: '(?=\S)'
+      set: type
+    - include: not-whitespace-illegal-pop
+  value-group-after:
+    - match: '\b(function|delegate)\b'
+      scope: keyword.other.d
+      set: [value-group-after, basic-type2, function-type]
+    - match: '\)'
+      scope: punctuation.section.group.end.d
+      set: value-group-after-parens
+    - match: '\b({{name}})\b'
+      scope: variable.parameter.d
+      set: value-group-after
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: [value-group-after-parens, function-argument, function-argument-or-type]
+    - match: '\.\.\.'
+      scope: keyword.operator.variadic.d
+    - include: not-whitespace-illegal
+  value-group-after-parens:
+    - include: function-attribute-in
+    - match: (?={)
+      set: [value-after, block-statement]
+    - match: '=>'
+      scope: keyword.operator.lambda.d
+      set: value
+    - match: (?=\S)
+      set: value-after
+  value-identifier:
+    - match: '\b{{name}}\b(?=\s*\()'
+      scope: variable.function.d
```

Comment:
This identifier, plus the parens and all parameter should be scoped with `meta.function-call`. The scopes `meta.function-call variable.function` are used for references in the Goto Definition popup and the Goto Reference command.